ivy_models/regnet/regnet.py

```diff
@@ -0,0 +1,178 @@
+import ivy
+import ivy_models
+from ivy_models.base import BaseModel, BaseSpec
+from .layers import BlockParams, SimpleStemIN, ResBottleneckBlock, AnyStage
+from typing import Optional, Callable
+
+from collections import OrderedDict
+
+
+class RegNetSpec(BaseSpec):
+    def __init__(
+        self,
+        block_params: BlockParams,
+        num_classes: int = 1000,
+        stem_width: int = 32,
+        stem_type: Optional[Callable[..., ivy.Module]] = None,
+        block_type: Optional[Callable[..., ivy.Module]] = None,
+        norm_layer: Optional[Callable[..., ivy.Module]] = None,
+        activation: Optional[Callable[..., ivy.Module]] = None,
+    ) -> None:
+        super(RegNetSpec, self).__init__(
+            block_params=block_params,
+            num_classes=num_classes,
+            stem_width=stem_width,
+            stem_type=stem_type,
+            block_type=block_type,
+            norm_layer=norm_layer,
+            activation=activation,
+        )
+
+
+class RegNet(BaseModel):
+    def __init__(
+        self,
+        block_params: BlockParams,
+        num_classes: int = 1000,
+        stem_width: int = 32,
+        stem_type: Optional[Callable[..., ivy.Module]] = None,
+        block_type: Optional[Callable[..., ivy.Module]] = None,
+        norm_layer: Optional[Callable[..., ivy.Module]] = None,
+        activation: Optional[Callable[..., ivy.Module]] = None,
+        spec=None,
+        v: ivy.Container = None,
+    ) -> None:
+        self.block_params = block_params
+        self.num_classes = num_classes
+        self.stem_width = stem_width
+        self.stem_type = stem_type
+        self.block_type = block_type
+        self.norm_layer = norm_layer
+        self.activation = activation
+
+        self.spec = (
+            spec
+            if spec and isinstance(spec, RegNetSpec)
+            else RegNetSpec(
+                block_params,
+                num_classes,
+                stem_width,
+                stem_type,
+                block_type,
+                norm_layer,
+                activation,
+            )
+        )
+        super(RegNet, self).__init__(v=v)
+
+    def _build(self, *args, **kwargs):
+        if self.stem_type is None:
+            self.stem_type = SimpleStemIN
+        if self.norm_layer is None:
+            self.norm_layer = ivy.BatchNorm2D
+        if self.block_type is None:
+            self.block_type = ResBottleneckBlock
+        if self.activation is None:
+            self.activation = ivy.ReLU
+
+        self.stem = self.stem_type(
+            3,  # width_in
+            self.stem_width,
+            self.norm_layer,
+            self.activation,
+        )
+
+        current_width = self.stem_width
+
+        blocks = []
+        for i, (
+            width_out,
+            stride,
+            depth,
+            group_width,
+            bottleneck_multiplier,
+        ) in enumerate(
+            zip(
+                self.block_params.widths,
```

Comment:
I see that this is the first time the class is used but there's no code like `self.block_params = BlockParams(...)`